examples/control_test.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from functools import partial
+
+from absl.testing import absltest
+import numpy as onp
+
+from jax import lax
+from jax import test_util as jtu
+import jax.numpy as np
+
+from examples import control
+
+from jax.config import config
+config.parse_flags_with_absl()
+FLAGS = config.FLAGS
+
+
+def one_step_lqr(dim, T):
+  Q = np.stack(T * (np.eye(dim),))
+  q = np.zeros((T, dim))
+  R = np.zeros((T, dim, dim))
+  r = np.zeros((T, dim))
+  M = np.zeros((T, dim, dim))
+  A = np.stack(T * (np.eye(dim),))
+  B = np.stack(T * (np.eye(dim),))
+  return control.LqrSpec(Q, q, R, r, M, A, B)
+
+
+def control_from_lqr(lqr):
+  T, dim, _ = lqr.Q.shape
+  dot = np.dot
+
+  def cost(t, x, u):
+    return (
+        dot(dot(lqr.Q[t], x), x) + dot(lqr.q[t], x) +
+        dot(dot(lqr.R[t], u), u) + dot(lqr.r[t], u) +
+        dot(dot(lqr.M[t], u), x))
+
+  def dynamics(t, x, u):
+    return dot(lqr.A[t], x) + dot(lqr.B[t], u)
+
+  return control.ControlSpec(cost, dynamics, T, dim, dim)
+
+
+def one_step_control(dim, T):
+
+  def cost(t, x, u):
+    return np.dot(x, x)
+
+  def dynamics(t, x, u):
+    return x + u
+
+  return control.ControlSpec(cost, dynamics, T, dim, dim)
+
+
+class ControlExampleTest(jtu.JaxTestCase):
```

Comment:
Are these tests actually executed?

Reply:
I think they are, because of [this .travis.yml line](https://github.com/google/jax/blob/f6e1d01f94936c992d9e63810eae5db69c06a026/.travis.yml#L43) that looks for tests in both the "tests" and "examples" directories.

To run them manually I did `env PYTHONPATH=. python examples/control_test.py`, and all passed! Woo!